network/fuzz.go

```diff
@@ -0,0 +1,46 @@
+// +build gofuzz
+package network
+
+import (
+	"bytes"
+	"fmt"
+)
+
+func Fuzz(data []byte) int {
```

Comment:
This function is only intended to be used by fuzz testing, right? I think it would make more sense to move it to `fuzz_test.go` in that case.